content: Remake Leo Rover specification (#44)
fictionlab/docs: docs/leo-rover/documentation/specification/specification.mdx

This is a technical specification of the Leo Rover 1.9. <br /> If you are
looking for specification of the older version of the Leo Rover, please visit:

<FlexTable width="100%">
  <FlexTableItem style={{ padding: '0 0.5rem 0 0.5rem' }}>
    <LinkButton docId="leo-rover/documentation/specification/specification-1.8" />
  </FlexTableItem>
  <FlexTableItem style={{ padding: '0 0.5rem 0 0.5rem' }}>
    <LinkButton docId="leo-rover/documentation/specification/specification-1.7" />
  </FlexTableItem>
</FlexTable>

</details>

---

## Introduction

Leo Rover is a compact, 4-wheeled, remote-controlled robot designed for indoor
and outdoor robotic project development, learning, and research applications. It
is equipped with a built-in computer, a high-resolution camera, and a powerful
battery, making it suitable for various tasks, including autonomous navigation,
obstacle detection, and remote monitoring.

### Main parameters

| Parameter                    |                Value                |
| :--------------------------- | :---------------------------------: |
| Dimensions (LxWxH)           |      424 mm x 445 mm x 303 mm       |

> | Weight                       |               ~6.5 kg               |

Reviewer: It's 7kg
```suggestion
| Weight                       |               ~7 kg               |
```

> | Weight                       |               ~6.5 kg               |

Reviewer: Replace every ~ with `$\approx$ `
```suggestion
| Weight                       |              $\approx$ 6.5 kg               |
```

| Maximum payload              |               ~5 kg\*               |
| Maximum linear speed         |              ~0.4 m/s               |
| Maximum angular speed        |              ~1 Rad/s               |

> | Estimated max. obstacle size |                ~70 m                |

Reviewer: Should be in millimeters:
```suggestion
| Estimated max. obstacle size |                ~70 mm                |
```

| IP protection rating         |                IP 55                |
| Operating temperature        |          -10 °C to +40 °C           |
| Run time                     | up to 4 hours with standard battery |
| Connection range             |             Up to 100 m             |

\* - on standard tires

#### Traction parameters

| Parameter                    |    Value    |
| :--------------------------- | :---------: |
| Track Width                  |   354 mm    |
| Wheelbase length             |   295 mm    |
| Ground clearance             |   108 mm    |
| Climb grade (no payload)     | 45° ( ~58%) |
| Climb grade with 5kg payload | 45° ( ~36%) |

> | Hill grade traversal         | 45° ( ~58%) |

Reviewer: This doesn't make sense

| ----------------: | :------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Built-in computer |    1     | Raspberry Pi 5 - A single-board computer developed by Raspberry Pi Ltd.<br /> <br /> Key features: <ul><li>Processor: Broadcom BCM2712, 2.4GHz quad-core 64-bit Arm Cortex-A76 CPU with cryptography extensions. </li><li> Memory: 4GB LPDDR4X-4267 SDRAM.</li><li>Storage: Micro SD card slot with support for high-speed SDR104 mode; PCIe 2.0 x1 interface for fast peripherals (requires separate 3 M.2 HAT or adapter).</li><li>Connectivity:<ul><li>Gigabit Ethernet (supports PoE+ with separate PoE+ HAT)</li><li>Dual-band 802.11ac Wi-Fi</li><li> Bluetooth 5.0 / Bluetooth Low Energy (BLE)</li><li>2 x USB 3.0 ports supporting simultaneous 5Gbps operation</li><li> 2 x USB 2.0 5 ports</li><li> 2 x micro-HDMI ports supporting resolutions up to 4kp60 with HDR support</li><li>2 x 4-lane MIPI camera/display transceivers</li><li>standard 40-pin GPIO header</li></ul></li></ul> |

Reviewer: Typo?
```suggestion
| Built-in computer |    1     | Raspberry Pi 5 - A single-board computer developed by Raspberry Pi Ltd.<br /> <br /> Key features: <ul><li>Processor: Broadcom BCM2712, 2.4GHz quad-core 64-bit Arm Cortex-A76 CPU with cryptography extensions. </li><li> Memory: 4GB LPDDR4X-4267 SDRAM.</li><li>Storage: Micro SD card slot with support for high-speed SDR104 mode; PCIe 2.0 x1 interface for fast peripherals (requires separate M.2 HAT or adapter).</li><li>Connectivity:<ul><li>Gigabit Ethernet (supports PoE+ with separate PoE+ HAT)</li><li>Dual-band 802.11ac Wi-Fi</li><li> Bluetooth 5.0 / Bluetooth Low Energy (BLE)</li><li>2 x USB 3.0 ports supporting simultaneous 5Gbps operation</li><li> 2 x USB 2.0 5 ports</li><li> 2 x micro-HDMI ports supporting resolutions up to 4kp60 with HDR support</li><li>2 x 4-lane MIPI camera/display transceivers</li><li>standard 40-pin GPIO header</li></ul></li></ul> |
```

  style={{
    width: '700px',
  }}
/>

### LeoCore controller

Leo Rover since version 1.8 is based on the LeoCore electronics board which,
together with the Raspberry Pi computer, controls all the Rover's
functionalities.

<ImageZoom
  src="/img/robots/leo/specification/LeoCore.webp"
  alt="LeoCore controller"
  width="1600"
  height="1120"
  figStyle={{
    width: 600,
  }}
/>

To make it easier, we listed all the interfaces used by Leo Rover as default.
Just to make sure you don't interfere with them when developing.

| Port                                    | Functionality                                                     |
| --------------------------------------- | ----------------------------------------------------------------- |
| Power input                             | to power the board                                                |
| RPi port                                | to power and RPi serial communication                             |
| LED output                              | to control the battery LED (to show system readiness)             |
| Motor output A, B, C & D (PWM H-bridge) | to power the rover motors and encodes                             |

> | ST-link/V2 6-pin port                   | used to flash firmware to the board using ST-link/V2 (optionally) |

Reviewer: ```suggestion
| 5-pin debug port                   | used to flash firmware to the board using ST-link/V2 (optionally) |
```

| -------------------------------: | :-----------------------: |
| Tire size (diameter x thickness) |    ~ 125 mm x ~ 70 mm     |
|                   Tire lock type |       non beadlock        |
|                      Tire insert | soft foam (non pneumatic) |
|               Wheel rim diameter |           98 mm           |

### Battery & charging

Leo Rover is powered by a 3S Li-Ion battery with a capacity of 7000 mAh. The
battery is equipped with an internal battery management system (BMS) that
provides protection against overcharging, over-discharging, and short circuits.
The battery is designed to be easily replaceable, allowing for quick swaps
during extended use.

Each battery pack has following characteristics:

|            Parameter |                  Value                   |
| -------------------: | :--------------------------------------: |
|              Voltage |             11.1 V (nominal)             |
|         Battery type |                  Li-Ion                  |
|         Battery cell |       18650 (Samsung INR18650-35E)       |
|             Capacity |       7 Ah (84,7 Wh - flight safe)       |

> |    Battery pack type |                   3S1P                   |

Reviewer: ```suggestion
|    Battery pack type |                   3S2P                   |
```

| Maximum output power |                  ~120 W                  |
|       Safety systems | Overcurrent, Reverse polarity protection |

#### Connectors

By default Leo Rover uses standard [WEIPU](https://www.weipuconnector.com/)
SP13-3 connectors for connecting battery, Main Electronics Box and other
possible addons.

<FlexTable>
  <FlexTableItem>

| Pin name | Cable color                   |
| -------- | ----------------------------- |
| DC-      | black                         |
| DC+      | red / black with white stripe |
| LED      | green                         |

  </FlexTableItem>
  <FlexTableItem style={{padding: '0 0.5rem 0 0.5rem'}}>

>     <WeipuDiagram

Reviewer: Left side is slightly cut off
```suggestion
    <WeipuDiagram
```
